File: Samples/Verbum/how-to-embed-a-barcode-into-a-document-e20025-11.1.6-/CS/Extensions/RichEditEx.cs
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.Internal;
using DevExpress.XtraRichEdit.Model;

namespace BizPad.Extensions {
    public class RichEditEx : RichEditControl, IInnerRichEditControlOwner {
        protected override void ActivateViewPlatformSpecific(RichEditView view) {
            base.ActivateViewPlatformSpecific(view);
            RemoveService(typeof(IFieldCalculatorService));
            AddService(typeof(IFieldCalculatorService), new FieldCalculatorServiceEx());
        }
    }
}

File: Samples/Verbum/how-to-embed-a-barcode-into-a-document-e20025-11.1.6-/VB/Extensions/RichEditEx.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.Internal
Imports DevExpress.XtraRichEdit.Model

Namespace BizPad.Extensions
	Public Class RichEditEx
		Inherits RichEditControl
		Implements IInnerRichEditControlOwner
		Protected Overrides Sub ActivateViewPlatformSpecific(ByVal view As RichEditView)
			MyBase.ActivateViewPlatformSpecific(view)
			RemoveService(GetType(IFieldCalculatorService))
			AddService(GetType(IFieldCalculatorService), New FieldCalculatorServiceEx())
		End Sub
	End Class
End Namespace
